hw3/01-characters.js

```diff
@@ -1,4 +1,32 @@
 // url for the Thrones API
-const url = 'https://thronesapi.com/api/v2/Characters';
+const url = "https://thronesapi.com/api/v2/Characters";
+const charactersElement = document.getElementById("characters");
 
-fetch(url);
+fetch(url)
+  .then((response) => response.json())
+  .then((characters) => {
+    characters.forEach((character) => {
+      const characterElement = document.createElement("div");
+      characterElement.classList.add("character");
+
+      const characterImage = document.createElement("img");
+      characterImage.src = character.imageUrl;
+      characterImage.alt = `${character.fullName} image`;
+      characterImage.classList.add("character-image");
+
+      const characterName = document.createElement("h2");
+      characterName.textContent = character.fullName;
+      characterName.classList.add("character-name");
+
+      const characterTitle = document.createElement("h3");
+      characterTitle.textContent = character.title;
+      characterTitle.classList.add("character-title");
+
+      characterElement.appendChild(characterImage);
+      characterElement.appendChild(characterName);
+      characterElement.appendChild(characterTitle);
+
+      charactersElement.appendChild(characterElement);
+    });
+  })
```

Comment:
This is a lot of code to put inside a forEach().

Consider creating an additional function which would be responsible for creating and appending elements "characters" to the DOM. That new function would have the dedicated task of taking the data and processing it into into elements to add to the DOM. The function name would match it's responsibility too, for example addCharactersToDom(). So in the future when someone wants to change how the data is displayed on the page it would be easy for them to know to focus on that function instead of looking through the forEach and modifying things in there. It's more about separation of concerns and making it readable.